tasks/coconut/qemu.py

```diff
@@ -0,0 +1,79 @@
+from invoke import task
+from os.path import join
+from subprocess import run
+from tasks.util.env import BIN_DIR, PROJ_ROOT, KATA_ROOT
+
+# refer to
+# https://github.com/coconut-svsm/svsm/blob/main/Documentation/docs/installation/INSTALL.md
+
+QEMU_IMAGE_TAG = "qemu-igvm-build"
+DATA_DIR = join(KATA_ROOT, "coconut", "qemu-svsm", "share")
+
+
+@task
+def build(ctx):
+    print(DATA_DIR)
+    docker_cmd = "docker build --build-arg QEMU_DATADIR={} -t {} -f {} .".format(
+        DATA_DIR,
+        QEMU_IMAGE_TAG,
+        join(PROJ_ROOT, "docker", "coconut", "qemu.dockerfile"),
+    )
+    run(docker_cmd, shell=True, check=True, cwd=PROJ_ROOT)
+
+    tmp_ctr_name = "tmp-qemu-igvm-run"
+    docker_cmd = "docker run -td --name {} {}".format(tmp_ctr_name, QEMU_IMAGE_TAG)
+    run(docker_cmd, shell=True, check=True)
+    copy_from_container(
+        tmp_ctr_name,
+        "/root/bin/qemu-svsm/bin/qemu-system-x86_64",
+        join(BIN_DIR, "qemu-system-x86_64-igvm"),
+    )
+    copy_from_container(tmp_ctr_name, f"{DATA_DIR}/.", DATA_DIR)
+    run("docker rm -f {}".format(tmp_ctr_name), shell=True, check=True)
+
+
+def copy_from_container(ctr_name, ctr_path, host_path):
```

Comment:
Does a method like this exist somewhere in `tasks/util`? If not, it would be good to add it and use it in `ovmf.py` too.

Reply:
Thank you, I refactored the files in tasks/coconut accordingly. Should we do this for all similar python scripts?